ostd/src/panicking.rs

```diff
@@ -36,30 +38,54 @@ pub fn panic_handler(info: &core::panic::PanicInfo) -> ! {
         abort();
     }
 
-    // If in ktest, we would like to catch the panics and resume the test.
+    use alloc::{boxed::Box, string::ToString};
+
+    use unwinding::panic::begin_panic;
+
     #[cfg(ktest)]
-    {
-        use alloc::{boxed::Box, string::ToString};
-
-        use unwinding::panic::begin_panic;
-
-        let throw_info = ostd_test::PanicInfo {
-            message: info.message().to_string(),
-            file: info.location().unwrap().file().to_string(),
-            line: info.location().unwrap().line() as usize,
-            col: info.location().unwrap().column() as usize,
-            resolve_panic: || {
-                IN_PANIC.store(false);
-            },
-        };
-        // Throw an exception and expecting it to be caught.
-        begin_panic(Box::new(throw_info.clone()));
-    }
-    early_println!("{}", info);
+    let throw_info = ostd_test::PanicInfo {
+        message: info.message().to_string(),
+        file: info.location().unwrap().file().to_string(),
+        line: info.location().unwrap().line() as usize,
+        col: info.location().unwrap().column() as usize,
+        resolve_panic: || {
+            IN_PANIC.store(false);
+        },
+    };
+    #[cfg(not(ktest))]
+    let throw_info = info.to_string();
+
+    // Throw an exception and expecting it to be caught.
+    begin_panic(Box::new(throw_info.clone()));
+
+    early_println!("Non-resettable panic! {}", info);
+
     print_stack_trace();
     abort();
 }
 
+/// Run user-injected code in a panic-safe way.
+pub(crate) fn catch_unwind_injected<F, R>(f: F) -> Option<R>
```

Comment:
This should be simply called `catch_unwind`. It has nothing to do with injected code. It could also be used to catch the panics in OSTD internal code.